src/Message.php

```diff
@@ -12,11 +12,13 @@
     /**
      * Message constructor.
      * @param string $payload
+     * @param string $iso8601
+     * @param string $encoding
      */
-    public function __construct($payload, $timestamp, $encoding)
+    public function __construct($payload, $iso8601, $encoding)
     {
         $this->payload = $payload;
-        $this->timestamp = new \DateTimeImmutable($timestamp);
+        $this->timestamp = new \DateTimeImmutable($iso8601);
```

Comment:
This could use the date passed on the constructor to create an immutable datetime object.

```
$this->timestamp = DateTimeImmutable::createFromMutable( $date );
```